src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -126,19 +166,17 @@ private void ReportUninitializedNonNullableReferenceTypeFields()
                 {
                     continue;
                 }
-                int fieldSlot = VariableSlot(field, thisSlot);
-                if (fieldSlot == -1 || !this.State.IsAssigned(fieldSlot))
+                if (getIsAssigned(walkerOpt, thisSlot, field))
                 {
-                    var symbol = member switch
-                    {
-                        FieldSymbol { AssociatedSymbol: PropertySymbol p } => p,
-                        _ => member
-                    };
-                    var location = (topLevelMethod.DeclaringSyntaxReferences.IsEmpty
-                        ? symbol // default constructor, use the field location
-                        : topLevelMethod).Locations[0];
-                    _diagnostics.Add(ErrorCode.WRN_UninitializedNonNullableField, location, symbol.Kind.Localize(), symbol.Name);
+                    continue;
                 }
+                var symbol = member switch
+                {
+                    FieldSymbol { AssociatedSymbol: PropertySymbol p } => p,
+                    _ => member
+                };
+                var location = getSymbolForLocation(walkerOpt, symbol).Locations.FirstOrDefault() ?? Location.None;
```

Comment:
What scenarios result in `location == Location.None`? It seems like something must have gone wrong somewhere in this case. If that's true consider adding a `Debug.Assert(location != null)` instead of `?? Location.None`.

Reply:
If this is actually the best thing to do, then consider using the `Locations.FirstOrNone()` extension.

Reply:
I don't believe there is a scenario today where `location == Location.None` but I added this just to be safe.

Will use `FirstOrNone()`, thanks.

---
In reply to: [307935470](https://github.com/dotnet/roslyn/pull/37507#discussion_r307935470) [](ancestors = 307935470)